### PR TITLE
Sec vendor on gulag now has ID and Cost requirements

### PR DESCRIPTION
**Pull request by FluffTheMoth01**
Signed-off-by: Space Prius <bubba041102@gmail.com>

## About The Pull Request
Fixes issue #45393

## Why It's Good For The Game
No more free sec equipment for gulag escapees.

## Changelog
:cl:
fix: The sec vendor in the gulag now actually has ID and cost requirements.
/:cl:

### CODE REVIEW

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -104,14 +106,14 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
...
 	power_change()
-
-	if(mapload) //check if it was initially created off station during mapload.
-		if(!is_station_level(z))
-			onstation = FALSE
-			if(circuit)
-				circuit.onstation = onstation //sync up the circuit so the pricing schema is carried over if it's reconstructed.
-	else if(circuit && (circuit.onstation != onstation)) //check if they're not the same to minimize the amount of edited values.
-		onstation = circuit.onstation //if it was constructed outside mapload, sync the vendor up with the circuit's var so you can't bypass price requirements by moving / reconstructing it off station.
+	if(!onstation_override) //overrides the checks if the variable is set to false.
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/vending/_vending.dm]:
```DM
if(onstation_override)
    onstation = TRUE
    return
//old, not nested code here
```
???

**Comment by ExcessiveUseOfCobblestone** [comment]
Really the onstation check should just also check for mining base areas rather than having to manually override any and all mining vendors.

File: code/modules/vending/_vending.dm (hunk 1 above)
Comment (SpacePrius) [reply code/modules/vending/_vending.dm]: It isn't supposed to be there, this is a weird git glitch

Comment (SpacePrius) [reply code/modules/vending/_vending.dm]: image

**Comment by SpacePrius** [comment]
I didn't do that because only one vendor needs to be overrided, seemed like too much work for what we get out of it.

File: code/modules/vending/_vending.dm (hunk 1 above)
Comment (SpacePrius) [reply code/modules/vending/_vending.dm]: This avoids running all that code for no reason.

**Comment by YakumoChen** [comment]
To be honest this shouldn't even be on the mine in the first place, the only other location for a secvend is secure in brig equipment room itself, I don't think a single other map has it in any other place. There definitely doesn't need to be a completely unguarded one on Lavaland.

**Comment by actioninja** [comment]
I'm not the biggest fan of the way this was achieved but it works.

**Pull request merged**